ggshield/cmd/sca/scan.py

```diff
@@ -57,14 +51,123 @@ def scan_diff_cmd(
 
 
 @scan_group.command(name="all")
+@click.argument(
+    "directory",
+    type=click.Path(exists=True, readable=True, path_type=Path, file_okay=False),
+    required=False,
+)
 @click.pass_context
-@display_sca_beta_warning
 def scan_all_cmd(
     ctx: click.Context,
     directory: Optional[Path],
     **kwargs: Any,
-) -> int:
+) -> Optional[SCAScanAllOutput]:
     """
     Scan a directory for SCA vulnerabilities.
     """
-    return 0
+    if directory is None:
+        directory = Path().resolve()
+
+    # Adds client and required parameters to the context
+    update_context(ctx)
+
+    result = sca_scan_all(ctx, directory)
+
+    # TODO, should be handled by an output_handler in the future
+    return result
+
+
+def update_context(ctx: click.Context) -> None:
+    """
+    Basic implementation, should be populated with required parameters later.
+    """
+    config: Config = ctx.obj["config"]
+    ctx.obj["client"] = create_client_from_config(config)
+
+    # Empty for now
+    ctx.obj["exclusion_regexes"] = set()
+
+
+def sca_scan_all(ctx: click.Context, directory: Path) -> Optional[SCAScanAllOutput]:
+    """
+    Scan an entire directory for SCA Vulnerabilities.
+
+    - List SCA related files with a first call to SCA compute files API
+    - Create a tar archive with the required files contents
+    - Launches the scan with a call to SCA public API
+    """
+    client = SCAClient(ctx.obj["client"])
+
+    try:
+        sca_filepaths = get_sca_scan_all_filepaths(
+            directory=directory,
+            exclusion_regexes=ctx.obj["exclusion_regexes"],
+            verbose=ctx.obj["config"].verbose,
+            client=client,
+        )
+    except client.ListScaFilesError:
```

Comment:
We don't want to catch the error this way, because we'll need to know an error happened when exiting, in order to set the exit code.

Reply:
We could as we return None in case of errors, but yes indeed we can just let it raise and catch it when we compute the return code from the command function.